src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h

```diff
@@ -123,6 +123,55 @@ namespace UiaOperationAbstraction
             }
         }
 
+        template<class TryBody, class CatchBody>
+        void Try(TryBody&& tryBody, CatchBody&& catchBody) const
+        {
+            if (m_useRemoteApi)
+            {
+                m_remoteOperation.TryBlock(std::forward<TryBody>(tryBody), std::forward<CatchBody>(catchBody));
+            }
+            else
+            {
+                try
+                {
+                    tryBody();
+                }
+                catch(std::exception /*e*/)
+                {
+                    catchBody();
+                }
+            }
+        }
+
+        template<class TryBody>
+        void Try(TryBody&& tryBody) const
+        {
+            if (m_useRemoteApi)
+            {
+                m_remoteOperation.TryBlock(std::forward<TryBody>(tryBody));
+            }
+            else
+            {
+                try
+                {
+                    tryBody();
+                }
+                catch(std::exception /*e*/)
+                {
+                }
+            }
+        }
+
+        std::optional<winrt::Microsoft::UI::UIAutomation::AutomationRemoteInt> GetCurrentFailureCode()

```

Comment:
For functions that don't need to live in the header, can we forward declare it here and actually define it in the cpp file?